src/folders_and_files/metadata.md

```diff
@@ -147,6 +147,124 @@ https://github.com/jeroen/jsonlite
     writeLines(data, file="myData.json")
 ```
 
+## Interoperability issues
+
+Many parts of JSON files are often loaded as
+[`structures`](https://nl.mathworks.com/help/matlab/ref/struct.html) by MATLAB /
+Octave, where a `key` in a JSON file becomes `fieldname` in that structure.
+
+Here is an example with a simple `example.json`
+
+```json
+{
+    "key": "value"
+}
+```
+
+loaded with bids-matlab
+
+```matlab
+>> json_content = bids.util.jsondecode('example.json')
+
+json_content =
+
+  struct with fields:
+
+    key: 'value'
+```
+
+There are however some strict rules for what makes a valid fieldnames in MATLAB
+and octave.
+
+Fieldnames must:
+
+-   start with a letter, otherwise assigning to that field will error
+-   contain only letters, numbers, and/or the underscore character, otherwise
+    assigning to that field will error, and
+-   must be no longer than `namelengthmax` (currently 63) characters, otherwise
+    you will receive a warning and the field name will be truncated
+
+If there are keys in your JSON that do not comply to those rules, they keys will
+be renamed when loading which can lead to some headaches down the line.
+
+For example when loading the `bad_keys.json`
+
+```json
+{
+    "@foo": "@foo",
+    "1": "1",
+    "/t": "/t",
+    "%f": "%f"
+}
+```
+
+We get some quite different fieldnames:
+
+```matlab
+>> json_content = bids.util.jsondecode('bad_keys.json')
+
+json_content =
+
+  struct with fields:
+
+    x_foo: '@foo'
+       x1: '1'
+      x_t: '/t'
+      x_f: '%f'
+```
+
+This can lead to some unexpected behavior if you did not know about this.
+
+If you load this `collision.json`
+
+```json
+{
+    "1": "1",
+    "x1": "x1",
+    "x_1": "x_1"
+}
+```
+
+and try to retrieve the value associated to the `key` `x1`, you will in fact be
+getting the value for the key `1`.
+
+```matlab
+>> json_content = bids.util.jsondecode('collision.json');
+>> json_content.x1
+
+       x1: '1'
+```
+
+**Why and when does this matter for BIDS?**
```

Comment:
This seems to be something that we are going to have to look at carefully and factor into HED tutorials. I am asking @dungscout96 and @arnodelorme to also look at this.  I think the three bullet points you have on lines 262-267 could be incorporated into the HED tutorials. I am assuming that not complying with these rules will still be okay with the bids validator?

Reply:
yup the validation of the content by the BIDS validator is pretty light weight as far as I know. so I don't think that something like is checked, @rwblair ? Though we had discussions about having several "depth" levels of validation but we are not yet there. 

But even if those "rules" were broken the dataset would still be valid.

But having something that at least throws a warning from the HED side when people try to generate sidecar template based on an events.tsv with problematic events name could be a nice thing to have.